frontend/src/routes/room/[id]/+page.svelte

```diff
@@ -29,10 +30,12 @@
   let roomInfo: RoomInfo | null = $state(null)
   let subtitles: Record<string, string | null> = $state({})
   let transientVideo: File | null = $state(null)
+  let typingStates: Record<string, boolean>[] = $state([])
```

Comment:
With the `timestamp` approach, this can be a `SvelteMap` mapping usernames to timestamps

Upon receiving a typing message, you:
- add the user to the map, then
- create a timeout for 5 seconds, wherein if the timestamp currently in `SvelteMap` === the timestamp when the timeout was set, then you can remove the username from the `SvelteMap`